src/lib/learn-client/api/progress/get-all-progress.ts

```diff
@@ -31,9 +31,23 @@ const getAllProgress = async ({
 	tutorialIds,
 	collectionIds,
 }: GetAllProgressOptions): Promise<GetAllProgressResult> => {
+	/**
+	 * TODO: figure out why this happens sometimes!
+	 * Maybe related to "queryClient.invalidate"? That's my best guess so far.
+	 */
+	const doesNotHaveAccessToken = typeof accessToken == 'undefined'
+	if (doesNotHaveAccessToken) {
+		console.log(
+			"getAllProgress: for some reason we're making a request even though we don't have an accessToken."
+		)
+	}
+
 	// Add query params to the URL, if applicable
 	let url = PROGRESS_API_ROUTE
-	const qs = queryString.stringify({ tutorialIds, collectionIds })
+	const qs = queryString.stringify(
+		{ tutorialIds, collectionIds },
+		{ arrayFormat: 'comma' }
```

Comment:
Fixes an issue with `queryString` where array params would be represented as multiple duplicate keys in the query string, which our API doesn't like. `comma` option makes em comma separated instead.